File: Strategy/Java/src/duck/RubberDuck.java
package eu.svetlin.design.patterns.strategy.duck;

import eu.svetlin.design.patterns.strategy.fly.FlyNoWay;
import eu.svetlin.design.patterns.strategy.quack.Squeak;

public class RubberDuck extends Duck {
    public RubberDuck() {
        super();
        flyBehavior = new FlyNoWay();
        quackBehavior = new Squeak();
    }

    @Override
    public void display() {
    System.out.println("I'm a rubber duck :)");
    }
}

File: Strategy/TypeScript/src/duck/RubberDuck.ts
import FlyNoWay from '../fly/FlyNoWay';
import Squeak from '../quack/Squeak';
import Duck from './Duck';

class RubberDuck extends Duck {
  constructor() {
    super();
    this.flyBehavior = new FlyNoWay();
    this.quackBehavior = new Squeak();
  }

  display() {
    console.log("I'm a rubber duck :)");
  }
}

export default RubberDuck;
